File: internal/model/redis.go
package model

type RedisConfig struct {
	Host     string `json:"host"`
	Port     int    `json:"port"`
	Username string `json:"username"`
	Password string `json:"password"`
	DB       int    `json:"db"`
}

type RedisKey struct {
	Key   string      `json:"key"`
	Type  string      `json:"type"`
	Value interface{} `json:"value"`
	TTL   int64       `json:"ttl"`
}

// ZSetMember 表示有序集合的成员
type ZSetMember struct {
	Member string  `json:"member"`
	Score  float64 `json:"score"`
}

File: web/src/types/redis.ts
export interface RedisConfig {
  host: string
  port: number
  password: string
  db: number
}

export interface RedisKey {
  key: string
  type: string
  ttl: number
  size: number
}

export interface KeyDetail {
  type: string
  value: any
  ttl: number
}

export interface CommandResult {
  success: boolean
  data: any
  error?: string
}

export interface RedisApi {
  executeCommand: (command: string) => Promise<string>
} 